kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -687,6 +696,39 @@ matchNormalizedAc pushValue wrapTermLike normalized1 normalized2
         _ -> empty
     lift $ Foldable.traverse_ pushValue concrete12
     lift $ Foldable.traverse_ pushValue abstractMerge
+  | otherwise = do
+      Monad.guard (null concrete1)
+      case opaque1 of
+          [frame1]
+            | [frame2] <- opaque2 -> do
+                push (Pair frame1 frame2)
+                let ac1 =
+                        wrapTermLike normalized1
+                            { Builtin.opaque = []
+                            }
+                    ac2 =
+                        wrapTermLike normalized2
+                            { Builtin.opaque = []
+                            }
+                push (Pair ac1 ac2)
+            | [element1] <- abstract1
+            , (key1, value1) <- Builtin.unwrapElement element1
+            , Just value2 <- Builtin.lookupSymbolicKeyOfAc key1 normalized2 -> do
+                lift $ pushValue (Pair value1 value2)
+                let ac2 =
+                        wrapTermLike
+                        $ Builtin.removeSymbolicKeyOfAc key1 normalized2
+                push (Pair frame1 ac2)
+            | [element1] <- abstract1
+            , (element2 : _) <- abstract2 -> do
```

Comment:
I think this is only drawing `element2` from the symbolic elements of the second Map, right? It should also cover the concrete elements.